### PR TITLE
fix(scanner): various CVSS fixes

### PR DESCRIPTION
**Pull request by RTann**
## Description

A few updates:

* Use `c.Score` for `vuln.Cvss`
* Support the extra, unloved CVSS metrics for CVSS 2 and 3. Support = ignore in this case.
* Do not bother fetching NVD vuln data if there are no CVSS metrics

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Deploy Central Services, wait for vulns, then run:

```
roxctl image scan -i docker.io/pcfautomation/pcf-automation-release-image@sha256:0d93f35623e865e1bcff8c1a53d2b8a24b2f8b3856a7c20b6a8f94bfc43f9876
```

You'll find Central does not panic (nor log a panic)

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by connorgorman — APPROVED** [review]

**Review by dcaravel — APPROVED** [review]
Overall LGTM, few comments for consideration.

File: scanner/enricher/nvd/nvd.go [hunk 1]
```diff
@@ -194,6 +197,11 @@ func (e *Enricher) FetchEnrichment(ctx context.Context, hint driver.Fingerprint)
 }
 
 func filterFields(cve *schema.CVEAPIJSON20CVEItem) *schema.CVEAPIJSON20CVEItem {
+	// Skip the CVE if there is no CVSS metrics to save space in the DB.
+	if cve.Metrics == nil || (cve.Metrics.CvssMetricV2 == nil && cve.Metrics.CvssMetricV30 == nil && cve.Metrics.CvssMetricV31 == nil) {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (RTann) [reply scanner/enricher/nvd/nvd.go:203]: That's what we do today. I will move this check to later, though, so we can still get some fields out of NVD

**Comment by RTann** [comment]
@connorgorman @dcaravel I updated the PR to add a few more defensive measures, so I dismissed the reviews. Can you take another look?

File: pkg/scanners/scannerv4/convert.go [hunk 2]
```diff
@@ -117,21 +120,21 @@ func layerIndex(layerSHAToIndex map[string]int32, env *v4.Environment) *storage.
...
+		if !uniqueVulns.Add(id) {
+			// Already saw this vulnerability, so ignore it.
 			continue
 		}
 
-		if !uniqueVulns.Add(ccVuln.Name) {
-			// Already added this vulnerability, so ignore it.
+		ccVuln, ok := vulnerabilities[id]
+		if !ok {
+			log.Debugf("vuln ID %q from PackageVulnerabilities not found in Vulnerabilities, skipping", id)
```
Comment (RTann) [inline pkg/scanners/scannerv4/convert.go:137]: This is possible, as I've seen when testing. It's not necessarily unexpected, as we may have filtered out some incomplete vulns. Discussing offline

**Review by dcaravel — APPROVED** [review]

**Pull request merged**